sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCV2Suite.scala

```diff
@@ -1412,4 +1413,15 @@ class JDBCV2Suite extends QueryTest with SharedSparkSession with ExplainSuiteHel
       }
     }
   }
+
+  test("register dialect specific functions") {
+    H2Dialect.registerFunction("my_avg", IntegralAverage)
```

Comment:
let's add a try-catch-finally to clear the registered functions

Reply:
OK

Reply:
Seems it's hard to do `clearFunctions` for one test case only, how about we do the following:
1. in `beforeAll`, we register functions to `H2Dialect`
2. in `afterAll`, we clear functions in `H2Dialect`

The result is, the entire test suite will test against a JDBC catalog with a UDF. Other test suites will use a fresh SparkSession and instantiate new `JDBCTabeCatalog` instance and won't be affected.

Reply:
OK